tests/network/localnet/liblocalnet.py

```diff
@@ -1,15 +1,65 @@
+from collections.abc import Generator
+from contextlib import contextmanager
+
+import tests.network.libs.nodenetworkconfigurationpolicy as libnncp
 from libs.net import netattachdef
-from libs.net.vmspec import add_network_interface, add_volume_disk
+from libs.net.traffic_generator import Client, Server
+from libs.net.vmspec import IP_ADDRESS, add_network_interface, add_volume_disk, lookup_iface_status
 from libs.vm.affinity import new_pod_anti_affinity
 from libs.vm.factory import base_vmspec, fedora_vm
 from libs.vm.spec import CloudInitNoCloud, Interface, Metadata, Multus, Network
 from libs.vm.vm import BaseVirtualMachine, cloudinitdisk_storage
 from tests.network.libs import cloudinit
+from tests.network.libs.nodenetworkconfigurationpolicy import Resource
+
+_IPERF_SERVER_PORT = 5201
+NNCP_INTERFACE_TYPE_OVS_BRIDGE = "ovs-bridge"
+
+
+@contextmanager
+def running_localnet_vms(
+    vms: tuple[BaseVirtualMachine, BaseVirtualMachine],
+) -> Generator[tuple[BaseVirtualMachine, BaseVirtualMachine]]:
+    for vm in vms:
+        vm.start()
+    for vm in vms:
+        vm.wait_for_ready_status(status=True)
+        vm.vmi.wait_for_condition(condition=vm.Condition.Type.AGENT_CONNECTED, status=vm.Condition.Status.TRUE)
+    yield vms
+
+
+def create_traffic_server(vm: BaseVirtualMachine) -> Server:
+    return Server(vm=vm, port=_IPERF_SERVER_PORT)
+
 
-NETWORK_NAME = "localnet-network"
+def create_traffic_client(vms: tuple[BaseVirtualMachine, BaseVirtualMachine], network_name: str) -> Client:
+    vm_server, vm_client = vms
+    return Client(
+        vm=vm_client,
+        server_ip=lookup_iface_status(vm=vm_server, iface_name=network_name)[IP_ADDRESS],
+        server_port=_IPERF_SERVER_PORT,
+    )
+
+
+def additional_ovs_bridge_interface(bridge_name: str, worker_port_name: str) -> libnncp.Interface:
+    return libnncp.Interface(
+        name=bridge_name,
+        type=NNCP_INTERFACE_TYPE_OVS_BRIDGE,
+        ipv4=libnncp.IPv4(enabled=False),
+        ipv6=libnncp.IPv6(enabled=False),
+        state=Resource.Interface.State.UP,
```

Comment:
You better use `libnncp.NodeNetworkConfigurationPolicy.Interface.State.UP`, which has a more accurate context.